### ISSUE TITLE
Why `Set APIAuthenticationMethod`

### ISSUE THREAD

**Issue by freizl** [freizl/hoauth2#150]
e.g.

```
authGetBSWithAuthMethod ::
  -- | Specify the way that how to append the 'AccessToken' in the request
  Set.Set APIAuthenticationMethod ->
  -- | HTTP connection manager.
  Manager ->
  AccessToken ->
  URI ->
  -- | Response as ByteString
  ExceptT BSL.ByteString IO BSL.ByteString
```

Why would need to be Set (to support multiple way append access token in one request) instead of just `APIAuthenticationMethod`?

Likely in spec. find it out and add comment to the code or refactor to remove `Set`

**Comment by freizl** [comment]
From https://www.rfc-editor.org/rfc/rfc6750#section-2

> This section defines three methods of sending bearer access tokens in
>   resource requests to resource servers.  Clients MUST NOT use more
>   than one method to transmit the token in each request.

Seems wrong to accept an `Set`

**Comment by freizl** [comment]
- fixes in 7fa768d451cf10e09b317e6b461a8bb6689481d4
- probably because `authPostBS1` does support adding credential in multiple places and was trying to avoid breaking the contract. But dont remember any OAuth2 provider does require this.
